### PR TITLE
address issue 2780 and 2781 by adding a DECOY weapon type, weapon dat…

### PR DESCRIPTION
**Pull request by zhexu14**
This PR addresses https://github.com/dcs-liberation/dcs_liberation/issues/2780 and https://github.com/dcs-liberation/dcs_liberation/issues/2781.

To address https://github.com/dcs-liberation/dcs_liberation/issues/2780, this PR:
- Introduces a new weapon type, DECOY
- Creates the appropriate weapon information in *resources/weapons* for TALD
- Sets the correct weapon release type at the SEAD ingress waypoint when weapons of DECOY type are loaded

To address https://github.com/dcs-liberation/dcs_liberation/issues/2781, this PR:
- Stops the AttackGroup task after 120 seconds (somewhat arbitrary but seems to work)
- Adds a SwitchWaypoint task to skip the target waypoint and proceed directly to the Split waypoint
Both of the above changes are required to work around the lack of an appropriate RTB when out of ammo option in DCS

This PR has been tested by:
1. Running a mission from the following Liberation save in the development build, and confirming that the F14 flight does not release the decoys and overflies the target.
[test.liberation.zip](https://github.com/dcs-liberation/dcs_liberation/files/11306535/test.liberation.zip)

2. Running the same mission generated from this PR and confirming that the F14 flight 1)  releases the decoys and 2) does not overfly the target.
3. Running the same mission 1) from the Turn start time 2) after fast forwarding to a) just after take off, b) between Hold and Join points, c) just before the Ingress point and d) at the Ingress point. 

As an aside, the other SEAD flight in the package (Hornets with ARMs) also behaves weirdly, but that's probably another issue & PR.

### CODE REVIEW

**Comment by Starfire13** [comment]
Rather than doing the "stop the AttackGroup task after 120 seconds" thing, why not just remove the waypoint over the target for flights carrying TALDs? The failure to RTB when out of ammo only occurs with TALD flights (flights that carry other weapons RTB just fine), and TALD flights do not need a waypoint over the target (which was added for flights carrying ARMs when Skynet is enabled).

**Comment by zhexu14** [comment]
> Rather than doing the "stop the AttackGroup task after 120 seconds" thing, why not just remove the waypoint over the target for flights carrying TALDs? The failure to RTB when out of ammo only occurs with TALD flights (flights that carry other weapons RTB just fine), and TALD flights do not need a waypoint over the target (which was added for flights carrying ARMs when Skynet is enabled).

The motivation for the "skip waypoint" task rather than having Liberation not plan the target waypoint in the first place is due to the way Liberation goes about building a flight. From what I can see in the code, the logic of having Departure->Hold->Nav->Join->Ingress->Targets->Split->Refuel->Nav->Arrival->Divert->Bullseye as waypoints as well as how the target waypoints are defined is shared between a whole bunch of mission types e.g. SEAD, DEAD, OCA, Strike. Carving out SEAD when equipped with decoys would break that model. 

A couple of different ways of injecting different logic (new mission type, mission variants) were discussed in the original issue. However, from what I can tell, we're basically trying to work around a gap in DCS mission building, specifically the ability to specify RTB when out of decoys. What this PR is trying to do is put a localized patch to fix the behavior, until (hopefully) ED closes this gap without overly changing Liberation waypoint building logic.

Having said all of the above, I'm not overly familiar with the Liberation code, so if there's a better way to do this happy to give it a try :)

**Comment by Starfire13** [comment]
Ah, I had thought it would be simpler than that to use a conditional statement to skip (or remove afterwards) that waypoint for a TALD flight rather than carving out the TALD code into its own branch, but if that's not the case, then that makes sense.

File: game/data/weapons.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ class WeaponType(Enum):
     ARM = "ARM"
     LGB = "LGB"
     TGP = "TGP"
+    DECOY = "DECOY"
```
Comment (DanAlbert) [inline game/data/weapons.py]: `DECOY = "decoy"`. The others are all caps because they're acronyms :)

**Comment by DanAlbert** [comment]
Sorry, I meant to get to this today but got sidetracked. I should have time tomorrow.

Would you mind splitting it into 2 PRs? I'd have merged the fix for #2780 right now because it's simple enough, but I can't merge half a PR and I want more time to think about the other half.

**Comment by zhexu14** [comment]
> Sorry, I meant to get to this today but got sidetracked. I should have time tomorrow.
> 
> Would you mind splitting it into 2 PRs? I'd have merged the fix for #2780 right now because it's simple enough, but I can't merge half a PR and I want more time to think about the other half.

No worries, will split into two PRs tomorrow.

File: game/data/weapons.py (hunk 1 above)
Comment (zhexu14) [reply game/data/weapons.py]: Ah good point, fixed :)

**Comment by zhexu14** [comment]
> Sorry, I meant to get to this today but got sidetracked. I should have time tomorrow.
> 
> Would you mind splitting it into 2 PRs? I'd have merged the fix for #2780 right now because it's simple enough, but I can't merge half a PR and I want more time to think about the other half.

Actually, got a chance to split out the PRs. This PR now only addresses #2780 and the fix for #2781 has been split out to https://github.com/dcs-liberation/dcs_liberation/pull/2814, which has been branched off this PR.

**Pull request merged**